From 4ebb92130302f61976100c4df27fdca3cc7328d8 Mon Sep 17 00:00:00 2001
From: Anton Fedchin <anightik@gmail.com>
Date: Sat, 17 Jun 2017 10:55:22 +0300
Subject: [PATCH] fix deprecated api

---
 include/freetype/config/ftstdlib.h | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/include/freetype/config/ftstdlib.h b/include/freetype/config/ftstdlib.h
index 05a4845..47dd6b0 100644
--- a/include/freetype/config/ftstdlib.h
+++ b/include/freetype/config/ftstdlib.h
@@ -143,7 +143,11 @@
 
 
 #define ft_strtol  strtol
-#define ft_getenv  getenv
+#if defined(WINAPI_FAMILY) && WINAPI_FAMILY == WINAPI_FAMILY_DESKTOP_APP
+# define ft_getenv  getenv
+#else
+# define ft_getenv(x)  NULL
+#endif
 
 
   /**********************************************************************/
-- 
2.10.2.windows.1

