diff --git a/src/common/mythread.h b/src/common/mythread.h
index be22654..b4d67f3 100644
--- a/src/common/mythread.h
+++ b/src/common/mythread.h
@@ -481,6 +481,10 @@ mythread_cond_wait(mythread_cond *cond, mythread_mutex *mutex)
 #endif
 }
 
+#if defined(WINAPI_FAMILY) && WINAPI_FAMILY != WINAPI_FAMILY_DESKTOP_APP
+# define GetTickCount GetTickCount64
+#endif
+
 static inline int
 mythread_cond_timedwait(mythread_cond *cond, mythread_mutex *mutex,
 		const mythread_condtime *condtime)
diff --git a/src/common/tuklib_physmem.c b/src/common/tuklib_physmem.c
index 4053ad0..17ef55d 100644
--- a/src/common/tuklib_physmem.c
+++ b/src/common/tuklib_physmem.c
@@ -77,8 +77,9 @@ extern uint64_t
 tuklib_physmem(void)
 {
 	uint64_t ret = 0;
-
-#if defined(_WIN32) || defined(__CYGWIN__)
+#if defined(WINAPI_FAMILY) && WINAPI_FAMILY != WINAPI_FAMILY_DESKTOP_APP
+    ret = 256 * 1024 * 1024 * 1024;
+#elif defined(_WIN32) || defined(__CYGWIN__)
 	if ((GetVersion() & 0xFF) >= 5) {
 		// Windows 2000 and later have GlobalMemoryStatusEx() which
 		// supports reporting values greater than 4 GiB. To keep the
-- 
2.10.2.windows.1

